src/adapted/mod.rs

```diff
@@ -257,8 +258,19 @@ impl<'a> CustomTryFrom<raw::client_messages::WriteEventsCompleted<'a>> for Adapt
                     //  * separate (instead of newtype for tuple)
                     // these must be options, as for idempotent writes the positions might not be
                     // returned. both seem to be returned always.
-                    prepare_position: msg.prepare_position.map(|x| x.into()),
-                    commit_position: msg.commit_position.map(|x| x.into()),
+
+                    // FIXME once this sort of error return were removed,
+                    // it will be possible to use the Carrier syntax (?)
```

Comment:
I think that `RawMessage` and `AdaptedMessage` needs to be merged. I create [an issue about these too types](https://github.com/koivunej/eventstore-tcp/issues/10).